File: api/app/schemas/access_request.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class AccessRequestCreate(BaseModel):
    purpose: str = Field(..., min_length=1, max_length=1000)
    role: str | None = Field(default=None, max_length=120)


class AccessRequestResponse(BaseModel):
    id: str
    user_id: str
    asset_id: str
    purpose: str
    role: str | None
    status: Literal["pending", "approved", "rejected"]
    decision_reason: str
    decided_at: datetime | None
    created_at: datetime


class AccessRequestDecision(BaseModel):
    reason: str = Field(default="", max_length=1000)

File: web/src/types/access-request.ts
export type AccessRequestStatus = "pending" | "approved" | "rejected";

export type AccessRequest = {
  id: string;
  user_id: string;
  asset_id: string;
  purpose: string;
  role: string | null;
  status: AccessRequestStatus;
  decision_reason: string;
  decided_at: string | null;
  created_at: string;
};

export type AccessRequestPayload = {
  purpose: string;
  role?: string | null;
};
